components/automate-deployment/pkg/majorupgradechecklist/checklist_manager.go

```diff
@@ -0,0 +1,214 @@
+package majorupgradechecklist
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	"github.com/chef/automate/components/automate-deployment/pkg/cli"
+	"github.com/chef/automate/components/automate-deployment/pkg/manifest"
+	"github.com/pkg/errors"
+)
+
+type PreChecklistManager interface {
+	RunChecklist() error
+}
+
+type PerCheckList struct {
```

Comment:
struct not being used, please remove

Reply:
Anyone in the future wants to add a pre-checklist in a JSON file then this PerCheckList struct can be used